preview-env/clean/action.yml

```diff
@@ -0,0 +1,444 @@
+              --indent 0 \
+              '. += [{pullRequestId: $pr_id, pullRequestNumber: $pr_number, pullRequestActors: $pr_actors, lastDeployedAt: $updated_at, deploymentId: $deployment_id}]'
+          )
+        fi
+      done
+
+      echo results=$results >> $GITHUB_OUTPUT
+    shell: bash
+  - name: Process active preview environments
+    env:
+      DRY_RUN: ${{ inputs.dry-run }}
+      GH_TOKEN: ${{ inputs.token || github.token }}
+      LABELS: ${{ inputs.labels }}
+      PULL_REQUESTS: ${{ steps.pull-requests.outputs.results }}
+      REPOSITORY_NAME: ${{ steps.inputs.outputs.repository-name }}
+      REPOSITORY_OWNER: ${{ steps.inputs.outputs.repository-owner }}
+      SHUTDOWN_MESSAGE: ${{ inputs.shutdown-message }}
+      TTL: ${{ steps.inputs.outputs.ttl-seconds }}
+      WARNING_MESSAGE: ${{ inputs.warning-message }}
+      WARNING_TTL: ${{ steps.inputs.outputs.warning-ttl-seconds }}
+    run: |
+      set -e
+
+      # Get current date (UTC) in seconds
+      now=$(date --utc '+%s')
+
+      # Candidate Pull Requests with active preview environment(s)
+      pull_requests=$(echo $PULL_REQUESTS | jq -c '.[]')
+
+      # Get the Id(s) of the preview environment label(s)
+      label_ids=""
+      for label in ${LABELS//,/ }; do
+        label_ids="$label_ids "$(
+          gh api graphql \
+            --paginate \
+            --field owner=$REPOSITORY_OWNER \
+            --field repo=$REPOSITORY_NAME \
+            --field label=$label \
+            --field query=@${{ github.action_path }}/graphql/query_get_label_id.gql \
+            --jq '.data.repository.label.id'
+        )
+      done
+
+      # Process each Pull Request with active preview environment(s)
+      for pr in $pull_requests; do
+        # Get Pull Request infos
+        deployment_id=$(echo $pr | jq -r .deploymentId)
+        pr_id=$(echo $pr | jq -r .pullRequestId)
+        pr_last_deployed_at=$(
+          date \
+            --date="$(echo $pr | jq -r .lastDeployedAt)" \
+            +%s
+        )
+        pr_number=$(echo $pr | jq -r .pullRequestNumber)
+        pr_actors=$(echo $pr | jq -r .pullRequestActors)
+
+        # Compute lifetime
+        # Time between now and last deployment activity, regardless of environment if more than one
+        lifetime=$(($now-$pr_last_deployed_at))
+
+        # Get existing comment if any
+        unique_comment_tag="<!-- preview-env -->"
+        existing_comment_id=$(
+          gh api \
+            --paginate \
+            /repos/$REPOSITORY_OWNER/$REPOSITORY_NAME/issues/$pr_number/comments | \
+          jq \
+            --arg unique_comment_tag "$unique_comment_tag" \
+            -r \
+            '(.[] | select(.body | contains($unique_comment_tag))).id'
+        )
+
+        # Comment content to be created or updated if necessary
+        new_comment_body=""
+
+        echo "Processing PR #$pr_id (lifetime=${lifetime}s,ttl=${TTL}s,warning-ttl=${WARNING_TTL}s)"
```

Comment:
any chance we could use the PR number instead?
`#PR_kwDOA5eBQs5YH_uf` vs `#1342` would help with debugging.
The id is part of the source code of a page but if you have 10+ preview-envs it will be easier to use the PR number for debugging / plausibility checks.

Reply:
Good point. I will switch to PR number.